### PR TITLE
fix: filter out null request parameters

### PR DESCRIPTION
**Pull request by P0lip**
Fixes https://github.com/stoplightio/studio-internal/issues/1608

### CODE REVIEW

File: src/oas3/transformers/request.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ export function translateToRequest(
   };
 
   for (const parameter of parameters) {
+    if (!isObject(parameter)) continue; // just in case (handled by `getOasParameters` already)
```
Comment (XVincentX) [inline src/oas3/transformers/request.ts]: Hmm, if so — can't we remove it?

File: src/oas/accessors.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
-import { unionBy } from 'lodash';
+import { isObject, unionBy } from 'lodash';
 
 export function getOasParameters<ParamType extends { name: string; in: string }>(
```
Comment (XVincentX) [inline src/oas/accessors.ts]: This should probably become `getValidOasParameters` – but we can keep it as it is.

**Comment by P0lip** [comment]
> Looks good but can you do the same of the oas2 parameters?

@lottamus they seem to use the same accessor, hence the oas2 case should be covered as well.
I'll add a test proving that.

File: src/oas3/transformers/request.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ import {
   IHttpQueryParam,
   IMediaTypeContent,
 } from '@stoplight/types';
-import { map, omit, pickBy } from 'lodash';
+import { isObject, map, omit, pickBy } from 'lodash';
```
Comment (lottamus) [inline src/oas3/transformers/request.ts]:
```suggestion
import { map, omit, pickBy } from 'lodash';
```

File: src/oas2/__tests__/operation.test.ts [hunk 4]
```diff
@@ -90,4 +91,52 @@ describe('transformOas2Operation', () => {
...
+      paths: {
+        '/users/{userId}': {
+          get: {
+            parameters: [
+              {
+                in: 'header',
+                name: 'name',
+                schema: {
+                  type: 'string',
+                },
```
Comment (lottamus) [inline src/oas2/__tests__/operation.test.ts]: OpenAPI v2 parameters don't have a "schema" property if you want to be technically correct 🤓 

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameters-definitions-object

```ts
type: 'string',
```

**Pull request merged**